### ISSUE TITLE
sonarqube-repair doesn't work at all on real projects.

### ISSUE THREAD

**Issue by fermadeiral** [SpoonLabs/sonarqube-repair#96]
The way we use sonarjava only works when the noncompliant line is marked with `// Noncompliant`, which is why our tests pass, but on real projects no issue will be found by sonarjava and consequently sonarqube-repair will not apply any change. I will search for a workaround, but I started to think that it's not a good idea to use sonarjava. Any help is welcome.

**Comment by henry-lp** [comment]
I also suspected from before when I start working with issue #75 , but could not be sure since I did not investigated this in details. Uptil now I think our spoon-only tool can detect the bug just as fine as `sonarjava`, even better since we can precisely sort out the exact cases which we can repair instead of depending on a third party. Currently I fully believe that almost every detection in sonarjava is also feasible using only spoon, unless their static code parser is somehow much superior than ours.
